core/src/main/java/org/elasticsearch/transport/netty/MessageChannelHandler.java

```diff
@@ -226,14 +226,17 @@ protected String handleRequest(Channel channel, Marker marker, StreamInput buffe
         transportServiceAdapter.onRequestReceived(requestId, action);
         NettyTransportChannel transportChannel = null;
         try {
-            transport.inFlightRequestsBreaker().addEstimateBytesAndMaybeBreak(messageLengthBytes, "<transport_request>");
-            transportChannel = new NettyTransportChannel(transport, transportServiceAdapter, action, channel,
-                requestId, version, profileName, messageLengthBytes);
-
             final RequestHandlerRegistry reg = transportServiceAdapter.getRequestHandler(action);
             if (reg == null) {
                 throw new ActionNotFoundTransportException(action);
             }
+            long reservedBytes = 0;
+            if (reg.isCheckSizeLimit()) {
```

Comment:
@nik9000 Yes, you're right. That was just a misunderstanding. It makes total sense to use `addWithoutBreaking` here as certain actions can also lead to be pretty large request size (like the `TransportReplicationAction` that Jason has mentioned).